test/e2e/storage/provision.go

```diff
@@ -85,6 +75,17 @@ var _ = ginkgo.Describe("provision volumes with different volume modes from volu
 				}
 			}
 		}
+		for _, claim := range local.pvcs {
```

Comment:
Are these tests intended only to be run with CSI hostpath driver, or with any CSI driver?

For a general test that can be used against any csi driver, it needs to be able to support any ordering of deleting snapshots and pvcs. So what we've done with our generic test suite is to delete both PVCs and VolumeSnapshots in parallel, and then wait for both of them to complete.

cc @xing-yang